platforms/software/build-init/src/main/java/org/gradle/buildinit/plugins/internal/BuildContentGenerationContext.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.buildinit.plugins.internal;
+
+public class BuildContentGenerationContext {
+    private VersionCatalogDependencyRegistry versionCatalogDependencyRegistry;
+
+    public void setVersionCatalogDependencyRegistry(VersionCatalogDependencyRegistry versionCatalogDependencyRegistry) {
```

Comment:
🤔 Who needs to set this once the object is created? I think it would better be a final field set in the constructor.

Reply:
good point, changed